src/audio/eq_iir/eq_iir.c

```diff
@@ -710,288 +704,211 @@ static int eq_iir_verify_params(struct comp_dev *dev,
+	struct comp_data *cd = module_get_private_data(mod);
 
-	cd->eq_iir_func(dev, &source->stream, &sink->stream, frames);
+	comp_info(mod->dev, "eq_iir_get_config()");
 
-	buffer_stream_writeback(sink, sink_bytes);
-
-	/* calc new free and available */
-	comp_update_buffer_consume(source, source_bytes);
-	comp_update_buffer_produce(sink, sink_bytes);
+	return comp_data_blob_get_cmd(cd->model_handler, cdata, fragment_size);
 }
 
-/* copy and process stream data from source to sink buffers */
-static int eq_iir_copy(struct comp_dev *dev)
+static int eq_iir_process(struct processing_module *mod,
+			  struct input_stream_buffer *input_buffers, int num_input_buffers,
+			  struct output_stream_buffer *output_buffers, int num_output_buffers)
 {
-	struct comp_copy_limits cl;
-	struct comp_data *cd = comp_get_drvdata(dev);
-	struct comp_buffer *sourceb, *sinkb;
-	struct comp_buffer __sparse_cache *source_c, *sink_c;
+	struct comp_data *cd = module_get_private_data(mod);
+	struct audio_stream __sparse_cache *source = input_buffers[0].data;
+	struct audio_stream __sparse_cache *sink = output_buffers[0].data;
+	uint32_t frame_count = input_buffers[0].size;
 	int ret;
 
-	comp_dbg(dev, "eq_iir_copy()");
-
-	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer,
-				  sink_list);
-	source_c = buffer_acquire(sourceb);
-
 	/* Check for changed configuration */
 	if (comp_is_new_data_blob_available(cd->model_handler)) {
 		cd->config = comp_get_data_blob(cd->model_handler, NULL, NULL);
-		ret = eq_iir_setup(cd, source_c->stream.channels);
+		ret = eq_iir_setup(mod, source->channels);
 		if (ret < 0) {
-			comp_err(dev, "eq_iir_copy(), failed IIR setup");
-			buffer_release(source_c);
-
+			comp_err(mod->dev, "eq_iir_process(), failed IIR setup");
 			return ret;
+		} else if (cd->iir_delay_size) {
+			comp_dbg(mod->dev, "eq_iir_process(), active");
+			cd->eq_iir_func = eq_iir_find_func(source->frame_fmt, sink->frame_fmt,
+							   fm_configured,
+							   ARRAY_SIZE(fm_configured));
+		} else {
+			comp_dbg(mod->dev, "eq_fir_process(), pass-through");
+			cd->eq_iir_func = eq_iir_find_func(source->frame_fmt, sink->frame_fmt,
+							   fm_passthrough,
+							   ARRAY_SIZE(fm_passthrough));
 		}
 	}
 
-	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
-				source_list);
-	sink_c = buffer_acquire(sinkb);
+	cd->eq_iir_func(mod, &input_buffers[0], &output_buffers[0], frame_count);
+	module_update_buffer_position(&input_buffers[0], &output_buffers[0], frame_count);
+	return 0;
+}
 
-	/* Get source, sink, number of frames etc. to process. */
-	comp_get_copy_limits(source_c, sink_c, &cl);
+/**
+ * \brief Set EQ IIR frames alignment limit.
+ * \param[in,out] source Structure pointer of source.
+ * \param[in,out] sink Structure pointer of sink.
+ */
+static void eq_iir_set_alignment(struct audio_stream *source, struct audio_stream *sink)
+{
+	const uint32_t byte_align = 1;
+	const uint32_t frame_align_req = 1;
```

Comment:
why not set 	const uint32_t byte_align = 8;  const uint32_t frame_align_req = 2;?

Reply:
Thanks, it's fixed now.